nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/WEB-INF/partials/canvas/process-group-configuration.jsp

```diff
@@ -40,6 +40,12 @@
                         <div class="setting-name">Process group parameter context</div>
                         <div class="editable setting-field">
                             <div id="process-group-parameter-context-combo"></div>
+                            <div id="parameter-contexts-recursive-container">
+                                <div id="parameter-contexts-recursive" class="nf-checkbox checkbox-unchecked"></div>
+                                <div class="nf-checkbox-label">Apply recursively</div>
```

Comment:
The positioning of this checkbox seems out of place in comparison to the other form elements. One option could be to place it directly under the `Process Group Parameter Context` select field.

However, another complicating factor is that unlike the other settings, this setting is not persistent. In other words, if a user checks the box and clicks the `Apply` button, the box will be unchecked when returning to the screen. This seems like it could create some confusion for users, especially if the UI gives the impression that Parameter Context selection will always be applied to any nested Process Groups.

With that background, another option could be to move this checkbox field right above, or next to the `Apply` button. That would probably warrant changing the labeling, because then it would appear to apply to all settings, instead of just the Parameter Context.

Yet one more would be to introduce a new tab in the Process Group configuration dialog specific to Parameter Contexts. That would also help differentiate permanent settings from temporary options like this option to apply Parameter Contexts recursively.

Reply:
Thanks @exceptionfactory for your insights.

The original idea was to have a dedicated tab for Parameter Context however after a discussion this was dropped as the current UI implementation is not applicable for this. I think the main concern is that having Apply buttons at two different contexts (Process Group Configuration and Parameter Context) require some extra validations and checks which would probably introduce some regressions.

The agreement wast to add the Apply recursive button next to the Parameter Context selection field and as there are some more new additions to the Process Group Configuration screen, such as Per Process Group Logging and some stateless related configurations, have a separate PR for rearranging the screen.

The new screen planed to be similar to the Parameter Provider screen with multiple columns where configuration fields grouped together based on some logic which needs to be discussed as part of a different activity.

@mcgilman , @mtien-apache and @sardell please feel free to chime in and share your thoughts.